### ISSUE TITLE
SyntaxError: An invalid or illegal string was specified core.js:177:0

### ISSUE THREAD

**Issue by 5hanth** [daveyarwood/mantra#9]
Hi,

I get this error when 

``` clojure
(def sq (m/osc :type :square))
(m/play-note sq {:pitch 440 :duration 2000})
```

``` js
mantra.core.gain_ramp = (function mantra$core$gain_ramp(gain_node,level){
var time = gain_node.context.currentTime;
return gain_node.gain.exponentialRampToValueAtTime(level,(time + 0.1));
});
```

any pointers?

``` js
return gain_node.gain.exponentialRampToValueAtTime(level,(time + 0.1));
```

thanks.

**Comment by daveyarwood** [comment]
Hi @5hanth,

I'm not able to reproduce this error. Do you have an example project?

**Comment by 5hanth** [comment]
Hi @daveyarwood ,

To re-produce error:

``` sh
$ git clone https://5hanth@bitbucket.org/5hanth/test-mantra.git
$ cd test-mantra
$ lein figwheel    # http://localhost:3449
```

``` clojure
cljs.user=> (in-ns 'test-mantra.views)
nil
test-mantra.views=> (def sq (m/osc :type :square))
#'test-mantra.views/sq
test-mantra.views=> (m/play-note sq {:pitch 440 :duration 1000})
#object[DOMException SyntaxError: An invalid or illegal string was specified]
     mantra.core/gain-ramp (jar:file:/root/.m2/repository/mantra/mantra/0.5.1/mantra-0.5.1.jar!/mantra/core.cljs:91:44)
     mantra.core/play-note* (jar:file:/root/.m2/repository/mantra/mantra/0.5.1/mantra-0.5.1.jar!/mantra/core.cljs:228:6)
     mantra$core$play_note (jar:file:/root/.m2/repository/mantra/mantra/0.5.1/mantra-0.5.1.jar!/mantra/core.cljs:246:4)
test-mantra.views=> 
```

**Comment by daveyarwood** [comment]
Hmm... it's working for me, though I am seeing a stacktrace for a (presumably unrelated) [chronoid](http://github.com/daveyarwood/chronoid) bug that I just found and am working on at the moment. It's a minor bug that doesn't affect the basic functionality of mantra.

I do hear the square wave when I run `(m/play-note sq {:pitch 440 :duration 1000}` in the REPL, as well as when I browse to `file:///tmp/test-mantra/resources/public/index.html` and click the "make mantra sound" button. There are no JS console errors when I click the button.

```
Prompt will show when Figwheel connects to your application
To quit, type: :cljs/quit
cljs.user=> (in-ns 'test-mantra.views)
nil
test-mantra.views=> (def sq (m/osc :type :square))
#'test-mantra.views/sq
test-mantra.views=> (m/play-note sq {:pitch 440 :duration 1000})
#object[RangeError RangeError: Maximum call stack size exceeded]
RangeError: Maximum call stack size exceeded
    at Array.join (native)
    at Function.cljs.core.str.cljs$core$IFn$_invoke$arity$1 (file:///tmp/test-mantra/resources/public/js/compiled/out/cljs/core.js:9312:12)
    at Object.cljs$core$str [as str] (file:///tmp/test-mantra/resources/public/js/compiled/out/cljs/core.js:9294:22)
    at Object.cljs$core$pr_writer_impl [as pr_writer_impl] (file:///tmp/test-mantra/resources/public/js/compiled/out/cljs/core.js:29450:128)
    at cljs$core$pr_writer (file:///tmp/test-mantra/resources/public/js/compiled/out/cljs/core.js:29480:18)
    at file:///tmp/test-mantra/resources/public/js/compiled/out/cljs/core.js:29908:128
    at Object.cljs$core$pr_sequential_writer [as pr_sequential_writer] (file:///tmp/test-mantra/resources/public/js/compiled/out/cljs/core.js:29205:136)
    at Object.cljs$core$print_map [as print_map] (file:///tmp/test-mantra/resources/public/js/compiled/out/cljs/core.js:29897:18)
    at cljs.core.PersistentArrayMap.cljs$core$IPrintWithWriter$_pr_writer$arity$3 (file:///tmp/test-mantra/resources/public/js/compiled/out/cljs/core.js:30121:18)
    at Object.cljs$core$pr_writer_impl [as pr_writer_impl] (file:///tmp/test-mantra/resources/public/js/compiled/out/cljs/core.js:29380:12)
test-mantra.views=>
```

(The stacktrace at the end is due to the aforementioned chronoid bug.)

At this point, I'm thinking this may be a browser compatibility issue. It's working for me on Chrome. I'll try Firefox and Safari now.

**Comment by daveyarwood** [comment]
Aha! I'm getting the error in Firefox. Digging deeper...

**Comment by 5hanth** [comment]
awe.. thank you! You're right I'm on firefox. :)

**Comment by daveyarwood** [comment]
Fixed in mantra 0.5.3. It was an issue related to setting the gain node gain value to 0 upon creating oscillators -- more info on that [here](https://github.com/daveyarwood/mantra/blob/master/CHANGELOG.md#053-22016).

Thanks for reporting this!

**Comment by 5hanth** [comment]
:smile: :+1: :heart:
